images/virtualization-artifact/pkg/controller/vd/internal/source/http.go

```diff
@@ -0,0 +1,216 @@
+package source
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+
+	common2 "github.com/deckhouse/virtualization-controller/pkg/common"
+	"github.com/deckhouse/virtualization-controller/pkg/common/datasource"
+	vdutil "github.com/deckhouse/virtualization-controller/pkg/common/datavolume"
+	"github.com/deckhouse/virtualization-controller/pkg/common/vmd"
+	cc "github.com/deckhouse/virtualization-controller/pkg/controller/common"
+	"github.com/deckhouse/virtualization-controller/pkg/controller/importer"
+	"github.com/deckhouse/virtualization-controller/pkg/controller/service"
+	"github.com/deckhouse/virtualization-controller/pkg/controller/supplements"
+	"github.com/deckhouse/virtualization-controller/pkg/dvcr"
+	virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+)
+
+type HTTPDataSource struct {
+	statService     Stat
+	importerService Importer
+	diskService     *service.DiskService
+	dvcrSettings    *dvcr.Settings
+	logger          *slog.Logger
+}
+
+func NewHTTPDataSource(
+	statService Stat,
+	importerService Importer,
+	diskService *service.DiskService,
+	dvcrSettings *dvcr.Settings,
+) *HTTPDataSource {
+	return &HTTPDataSource{
+		statService:     statService,
+		importerService: importerService,
+		diskService:     diskService,
+		dvcrSettings:    dvcrSettings,
+		logger:          slog.Default().With("controller", vmd.ShortName, "ds", "http"),
+	}
+}
+
+func (ds HTTPDataSource) Sync(ctx context.Context, vd *virtv2.VirtualDisk) (bool, error) {
+	ds.logger.Info("Sync", "vd", vd.Name)
```

Comment:
`"ns", vd.GetNamespace()` ?

Reply:
Refactoring of the logs is not included here - I want to address it in a separate PR